AdblockPlusSafari/FilterList+Processing.m

```diff
@@ -0,0 +1,251 @@
+/*
+ * This file is part of Adblock Plus <https://adblockplus.org/>,
+ * Copyright (C) 2006-2015 Eyeo GmbH
+ *
+ * Adblock Plus is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * Adblock Plus is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Adblock Plus.  If not, see <http://www.gnu.org/licenses/&gt.
+ */
+
+#import "FilterList+Processing.h"
+
+#import "AdblockPlus.h"
+
+#include <yajl/yajl_parse.h>
```

Comment:
JSON of filter list is processed and important keys are extracted here. The method server also as the first JSON validation. It recognizes both filter list formats.

This code is being executed in host application, but I decided to stick with SAX-like parse, since it is more efficient and I have experience with that.